ssd1306/i2c_ssd1306.c

```diff
@@ -0,0 +1,375 @@
+	i2c_smbus_write_byte_data(drv_client, CMD_ADDR, 0);
+	i2c_smbus_write_byte_data(drv_client, CMD_ADDR, (LCD_WIDTH - 1));
+
+	/* Set page start / end */
+	i2c_smbus_write_byte_data(drv_client, CMD_ADDR, PAGE_ADDR);
+	i2c_smbus_write_byte_data(drv_client, CMD_ADDR, 0);
+	i2c_smbus_write_byte_data(drv_client, CMD_ADDR, (LCD_HEIGHT / 8 - 1));
+
+	i2c_smbus_write_byte_data(drv_client, CMD_ADDR, COM_SCAN_DEC);
+
+	i2c_smbus_write_byte_data(drv_client, CMD_ADDR, SET_CONTRAST);
+	/* Max contrast */
+	i2c_smbus_write_byte_data(drv_client, CMD_ADDR, 0xFF);
+
+	/* set segment re-map 0 to 127 */
+	i2c_smbus_write_byte_data(drv_client, CMD_ADDR, 0xA1);
+	i2c_smbus_write_byte_data(drv_client, CMD_ADDR, SET_NORMAL_DISPLAY);
+	/* set multiplex ratio(1 to 64) */
+	i2c_smbus_write_byte_data(drv_client, CMD_ADDR, 0xA8);
+	i2c_smbus_write_byte_data(drv_client, CMD_ADDR, (LCD_HEIGHT - 1));
+	/* 0xA4 => follows RAM content */
+	i2c_smbus_write_byte_data(drv_client, CMD_ADDR, 0xA4);
+	i2c_smbus_write_byte_data(drv_client, CMD_ADDR, SET_DISPLAY_OFFSET);
+	i2c_smbus_write_byte_data(drv_client, CMD_ADDR, 0x00);
+
+	i2c_smbus_write_byte_data(drv_client, CMD_ADDR, SET_DISPLAY_CLOCK_DIV);
+	i2c_smbus_write_byte_data(drv_client, CMD_ADDR, 0x80);
+
+	i2c_smbus_write_byte_data(drv_client, CMD_ADDR, SET_PRECHARGE);
+	i2c_smbus_write_byte_data(drv_client, CMD_ADDR, 0x22);
+
+	i2c_smbus_write_byte_data(drv_client, CMD_ADDR, SET_COM_PINS);
+	i2c_smbus_write_byte_data(drv_client, CMD_ADDR, 0x12);
+
+	i2c_smbus_write_byte_data(drv_client, CMD_ADDR, SET_VCOM_DETECT);
+	i2c_smbus_write_byte_data(drv_client, CMD_ADDR, 0x20);
+
+
+	i2c_smbus_write_byte_data(drv_client, CMD_ADDR, CHARGE_PUMP);
+	i2c_smbus_write_byte_data(drv_client, CMD_ADDR, 0x14);
+
+	i2c_smbus_write_byte_data(drv_client, CMD_ADDR, DISPLAY_ON);
+
+	return 0;
+}
+
+static inline void set_dirty(struct my_device *md)
+{
+	atomic_inc(&(md->dirty));
+}
+
+static ssize_t i2c_ssd1306_write(struct fb_info *info,
+		const char __user *buf, size_t count, loff_t *ppos)
+{
+	struct i2c_client *drv_client;
+	struct my_device *md = info->par;
+	int cnt;
+
+	if (!md)
+		return -EINVAL;
+
+	drv_client = md->client;
+	dev_info(&drv_client->dev, "%s: enter\n", __func__);
+
+	cnt = fb_sys_write(info, buf, count, ppos);
+
+	if (cnt < 0)
+		return cnt;
+
+	set_dirty(md);
+
+	return cnt;
+}
+
+static int i2c_ssd1306_blank(int blank_mode, struct fb_info *info)
+{
+	struct i2c_client *drv_client;
+	struct my_device *md = info->par;
+
+	if (!md)
+		return -EINVAL;
+
+	drv_client = md->client;
+	dev_info(&drv_client->dev, "%s: enter\n", __func__);
+	return -1;
+}
+
+static void i2c_ssd1306_fillrect(struct fb_info *info,
+		const struct fb_fillrect *rect)
+{
+	struct i2c_client *drv_client;
+	struct my_device *md = info->par;
+
+	if (!md)
+		return;
+
+	drv_client = md->client;
+	dev_info(&drv_client->dev, "%s: enter\n", __func__);
+}
+
+static void i2c_ssd1306_copyarea(struct fb_info *info,
```

Comment:
Why have you desided to refuse of these methods' functional and just use write method?

Reply:
I did the minimum possible driver functionality for my purposes (vizualization app)